pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -1585,18 +1595,34 @@ func (r *ReconcileInfinispan) serviceExternal(m *infinispanv1.Infinispan) *corev
 			Namespace: m.ObjectMeta.Namespace,
 			Labels:    lsService,
 		},
-		Spec: corev1.ServiceSpec{
+	}
+
+	if externalServiceType == corev1.ServiceTypeLoadBalancer {
+		externalService.Spec = corev1.ServiceSpec{
+			Type:     externalServiceType,
+			Selector: lsPodSelector,
+		}
+	} else if externalServiceType == corev1.ServiceTypeNodePort {
+		var portNum int32
+		if m.Spec.Expose.Ports != nil || len(m.Spec.Expose.Ports) > 0 {
+			portNum = m.Spec.Expose.Ports[0].NodePort
+		} else {
+			portNum = 0
+		}
+		externalService.Spec = corev1.ServiceSpec{
 			Type:     externalServiceType,
 			Selector: lsPodSelector,
 			Ports: []corev1.ServicePort{
 				{
 					Port:       int32(11222),
 					TargetPort: intstr.FromInt(11222),
-					// Fix NodePort to match exported port in kind configuration
-					NodePort: int32(30222),
+					NodePort:   portNum,
 				},
 			},
-		},
+		}
+	} else {
+		// Service type currently unsupported
```

Comment:
This looks like the place for unsupported type log entry.

Reply:
I've added logging in the caller, we have no logger here